pkg/scheduler/cache/cache.go

```diff
@@ -679,11 +684,21 @@ func (sc *SchedulerCache) Snapshot() *schedulingapi.ClusterInfo {
 		RevocableNodes: make(map[string]*schedulingapi.NodeInfo),
 	}
 
+	// If nodeSelector == "", will match all nodes
+	selector, err := labels.Parse(strings.TrimSpace(sc.nodeSelector))
+	if err != nil {
+		panic(fmt.Errorf("the provided selector %q is not valid: %v", sc.nodeSelector, err))
+	}
+
```

Comment:
why not register this block to `nodeInformer` filter event handler >?

Reply:
Because we must sync all nodes, otherwise the cache may go wrong when adding nodes existing nodes to this scheduler.

We have met this problem in my company.

Reply:
I think we should sync all cluster nodes, which will more efficient to add nodes to the current scheduler. 

And the current scheduler should still take care of others nodes which hold some pods. 

If we do filter in `nodeInformer`, we will get the wrong node idle resources.